File: Practice(Beginner)/Python3/Transform_the_Expression.py
# Source: https://www.codechef.com/problems/ONP

t = int(input())
for test in range(t):
    s = input()
    opt = []
    res = []
    for i in range(len(s)):
        if(s[i] == '('):
            opt.append(s[i])
        elif(s[i] == '+' or s[i] == '-' or s[i] == '*' or s[i] == '/' or s[i] == '^'):
            opt.append(s[i])
        elif(s[i] == ')'):
            res.append(opt[len(opt)-1])
            del(opt[len(opt)-1])
            del(opt[len(opt)-1])
        else:
            res.append(s[i])
    for j in res:
        print(j,end="")
    print()

File: Practice(Beginner)/C++/Transform the Expression.cpp
// Source: https://www.codechef.com/problems/ONP

#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--){
        string s;
        cin >> s;
        char c;
        vector<char> opt;
        vector<char> res;
        for (int i = 0; i < s.length(); ++i) {
            if(s[i] == '('){
                opt.push_back(s[i]);
            }
            else if(s[i] == '+' or s[i] == '-' or s[i] == '*' or s[i] == '/' or s[i] == '^' ){
                opt.push_back(s[i]);
            }
            else if (s[i] == ')'){
                c = opt[opt.size()-1];
                opt.pop_back();
                opt.pop_back();
                res.push_back(c);
            } else{
                res.push_back(s[i]);
            }
        }
        for (int i = 0; i < res.size(); ++i) {
            cout << res[i];
        }
        cout << endl;
    }
}